dukan/src/main/kotlin/net/thechance/dukan/service/DukanProductService.kt

```diff
@@ -129,6 +161,29 @@ class DukanProductService(
         return product
     }
 
+    fun isProductFavorite(userId: UUID, productId: UUID): Boolean {
+        return favoriteProductRepository.findByUserIdAndProductId(userId, productId) != null
+    }
+
+    fun toggleFavoriteStatus(userId: UUID, productId: UUID): Boolean {
+        val favorite = favoriteProductRepository.findByUserIdAndProductId(userId, productId)
+        return if (favorite != null) {
+            favoriteProductRepository.deleteById(favorite.id)
+            false
+        } else {
+            createFavoriteEntry(userId, productId)
+        }
```

Comment:
I think you can make it like this:

```suggestion
        return if ( favoriteProductRepository.deleteById(favorite.id) > 0) {      
            false
        } else {
            createFavoriteEntry(userId, productId)
        }
```